tripy/tripy/frontend/trace/ops/storage.py

```diff
@@ -54,21 +55,40 @@ def __init__(
             assert not any([dtype, device]), "Internal usage: dtype/device are inherited from memref."
             self.dtype = mlir_utils.convert_runtime_dtype_to_tripy_dtype(self.data.dtype)
             self.shape = tuple(data.shape)
+            self.stride = utils.get_stride(self.shape, tuple(data.strides))
             self.device = tp_device("gpu") if data.address_space == runtime.PointerType.device else tp_device("cpu")
             self.has_memref = True
         elif common_utils.is_empty(data):
             # special case: empty tensor
             self.dtype = utils.default(dtype, datatype.float32)
             self.shape = tuple(utils.get_shape(data))
+            self.stride = utils.get_stride(self.shape)
             self.data = memref.create_empty_memref(shape=self.shape, dtype=self.dtype)
             self.device = utils.default(device, tp_device("gpu"))
             self.has_memref = True
         else:
             self.dtype = dtype if dtype else common_utils.get_element_type(data)
             self.shape = tuple(utils.get_shape(data))
+            self.stride = utils.get_stride(self.shape)
             self.device = utils.default(device, tp_device("gpu"))
             self.has_memref = False
 
+        if not utils.is_canonical_stride(self.shape, self.stride):
+            canonical_stride = utils.get_canonical_stride(self.shape)
+            error_message = (
+                f"Non-canonical strides detected:\n"
+                f"  Shape: {self.shape}\n"
+                f"  Current stride: {self.stride}\n"
+                f"  Expected canonical stride: {canonical_stride}\n"
+                f"Non-canonical strides are not supported for Tripy tensors. "
+                f"This usually occurs when the tensor is not contiguous in memory. "
+                f"To resolve this issue:\n"
+                f"1. Call .contiguous() or .ascontiguousarray() on your tensor before converting to a Tripy tensor.\n"
```

Comment:
We should check if the incoming tensor is from Pytorch/Jax/Numpy and suggest appropriate message